quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/HostnameV2PropertyMappers.java

```diff
@@ -41,6 +47,51 @@ public static void validateConfig(Picocli picocli) {
         if (!inUse.isEmpty()) {
             picocli.warn("Hostname v1 options %s are still in use, please review your configuration".formatted(inUse));
         }
+        boolean isProd = Environment.PROD_PROFILE_VALUE.equals(org.keycloak.common.util.Environment.getProfile());
+        boolean httpsEnabled = HttpPropertyMappers.isHttpsEnabled();
+        String host = Configuration.getConfigValue(HostnameV2Options.HOSTNAME).getValue();
+        if (host != null && validateFullHostname(httpsEnabled, isProd, host, picocli)) {
+            return;
+        }
+        if (httpsEnabled) {
+            // must be tls passthrough, but there's no way of knowing the intention
+            return;
+        }
+        String proxyHeaders = Configuration.getConfigValue(ProxyOptions.PROXY_HEADERS).getValue();
+        if (proxyHeaders != null) {
+            // must not be tls passthrough, but there's no way of knowing the intention
+            return;
+        }
+        if (isProd) {
+            String warning = CONTEXT_WARNING + " Also if you are using a proxy, requests from the proxy to the server will fail CORS checks with 403s because the wrong origin will be determined.";
```

Comment:
```suggestion
            String warning = CONTEXT_WARNING + " Also if you are using a proxy, requests from the proxy to the server will fail CORS checks with 403s because the wrong origin will be determined. Make sure `proxy-headers` are configured properly.";
```